### PR TITLE
Automate publishing of Github releases

### PR DESCRIPTION
**Pull request by owais**
**Description:**
Pushing release tags will automatically create Github releases with binaries and a checksum file as release artifacts. The release engineer will need to prepare release notes in advance and update the body of the automatically created release manually after the CI job finishes.

**Testing:**  Tested locally and on CI with my fork.

### CODE REVIEW

**Comment by bogdandrutu** [comment]
@owais looks like you have some specific setup that circleci does not trigger on the PRs that you posted, it works for everyone else, can you take a look.

**Comment by owais** [comment]
Yes, looking into the same. I think it might be because I also build my fork in CircleCI. Circle probably builds my branches there instead of here. Will disable my fork in Circle and see if that helps.

**Comment by bogdandrutu** [comment]
@owais works now :)

**Comment by tigrannajaryan** [comment]
I am not sure I fully understand the change.

Before this the releases looked like this:
1. Go to Github Releases page. 
2. Click "new release: button , type release notes and publish it. 
3. CircleCI builds and publishes to Dockerhub.

What changes after this PR?

File: Makefile [hunk 1]
```diff
@@ -132,10 +132,11 @@ install-tools:
 	go install github.com/pavius/impi/cmd/impi
 	go install github.com/securego/gosec/cmd/gosec
 	go install honnef.co/go/tools/cmd/staticcheck
+	go install github.com/tcnksm/ghr
 
 .PHONY: otelcol
 otelcol:
-	GO111MODULE=on CGO_ENABLED=0 go build -o ./bin/$(GOOS)_$(GOARCH)/otelcol $(BUILD_INFO) ./cmd/otelcol
+	GO111MODULE=on CGO_ENABLED=0 go build -o ./bin/otelcol_$(GOOS)_$(GOARCH) $(BUILD_INFO) ./cmd/otelcol
```
Comment (pjanotti) [inline Makefile]: Changing the path here requires changing the docker-component too

Comment (owais) [reply Makefile]: Updated makefile to use the correct binary name. Thanks

**Comment by owais** [comment]
@tigrannajaryan Not much changes. Before the workflow was something like this:

1. Create draft release with notes.
2. Create git tag and push tag to github
3. CircleCI automatically publishes docker images
4. Publish draft release

After this PR it'll be the following

1. Prepare draft release notes in GH/slack/local.
2. Create git tag and push to github.
3. CircleCI will publish images to docker hub, create a new release on Github for the tag, upload all binaries built to the CircleCI release as build artifacts along with a file containing checksum of each build.
4. Edit the automatically created release and paste release notes you prepared in step 1. 

The purpose was to automatically upload built binaries as release artifacts along with checksum information. The tool does not have a way to update an existing draft release with the artifacts. If we feel we really want to create draft release notes in GH before pushing the tag and then have Circle publish artifacts to the same, we can add support for it later.

**Pull request merged**